packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -726,12 +724,7 @@ export class ConnectionManager implements IConnectionManager {
 
         this.disconnectFromDeltaStream(disconnectMessage);
 
-        const canRetry = error !== undefined ? canRetryOnError(error) : true;
-
-        // If reconnection is not an option, close the DeltaManager
-        if (!canRetry) {
-            this.props.closeHandler(normalizeError(error, { props: fatalConnectErrorProp }));
-        } else if (this.reconnectMode === ReconnectMode.Never) {
```

Comment:
not sure why canRetry handling is removed. Is this intentional?
I believe nackHandler() workflow can result in non-retryable error. I.e. getNackReconnectInfo() & nackContent.code !== 403.

Reply:
Please read PR description

Reply:
Thanks! I'd suggest for nackHandler() to not even call this function in case of canRetry - false, and just add assert(canRetry, "all code paths reach here with retryable error") and remove any handling or events below - it feels clearly RE enforcing invariants and capturing any changes in workflows sooner.

Reply:
Might not be true though.  See error handler in documentDeltaConnection

Reply:
```
const err = this.createErrorObject("error", error, error !== "Invalid namespace");
```

Reply:
If my memory serves right, this case will result in failed connection, not disconnect. It happens before we get connect_document_success. I did not retraced code again, but I believe all reconnection flows were always passing canRetry = true except for nack path, so even if this error can result in disconnect, we are not handling it correctly.

Reply:
I think you're right. I want to start with the log and see if it bears out, I can follow up later and switch to assert. Not going to open an issue, I don't think it is critical to ensure it happens.